File: NadakkiMarketingSuite/backend/routes/scheduler.py
"""Scheduler API Routes."""
from fastapi import APIRouter
from ..services.scheduler_service import scheduler_service

router = APIRouter(prefix="/api/v1/scheduler", tags=["scheduler"])

@router.post("/start")
async def start_scheduler():
    return scheduler_service.start()

@router.post("/stop")
async def stop_scheduler():
    return scheduler_service.stop()

@router.get("/status")
async def get_status():
    return scheduler_service.get_status()

@router.get("/jobs")
async def list_jobs():
    return {"jobs": scheduler_service.get_jobs()}

File: nadakki-suite-enterprise/apps/dashboard/src/lib/api/scheduler.ts
import { apiClient } from './client';
export interface SchedulerStatus { running: boolean; jobs_count: number; active_jobs: number; next_scheduled: string | null; }
export interface ScheduledJob { id: string; name: string; status: string; cron_expression: string | null; last_run: string | null; }
export const schedulerApi = {
  getStatus: () => apiClient.get<SchedulerStatus>('/api/v1/scheduler/status'),
  listJobs: () => apiClient.get<ScheduledJob[]>('/api/v1/scheduler/jobs'),
  start: () => apiClient.post('/api/v1/scheduler/start', {}),
  stop: () => apiClient.post('/api/v1/scheduler/stop', {}),
};
